tox.ini

```diff
@@ -36,6 +33,41 @@ commands = flake8 {posargs}
 [testenv:venv]
 commands = {posargs}
 
+[pytest]
+filterwarnings =
```

Comment:
@jwodder is there a way to instruct pytest to error out instead of xfail while encountering nose-style parametric (generator) test? e.g. 
```
datalad/cli/tests/test_exec.py::test_call_from_parser_pos_arg_underscore XFAIL ([NOTRUN] yield...)
```
?

Reply:
Done.

Reply:
I am getting 
```
    raise _OptionError("unknown warning category: %r" % (category,)) from None
warnings._OptionError: unknown warning category: 'PytestCollectionWarning'
```

while trying to run tests now with pytest 6.2.5 . upgrade of pytest results in a new related crash:

```
$> python -m pytest --version      
pytest 7.1.1
ERROR: while parsing the following warning configuration:

  error:.*yield tests:PytestCollectionWarning

This error occurred:

Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad/venvs/dev3/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1638, in parse_warning_filter
    category: Type[Warning] = _resolve_warning_category(category_)
  File "/home/yoh/proj/datalad/datalad/venvs/dev3/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1677, in _resolve_warning_category
    cat = getattr(m, klass)
AttributeError: module 'builtins' has no attribute 'PytestCollectionWarning'
```